command/agent/auth/auth.go

```diff
@@ -139,6 +142,13 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) {
 			})
 			clientToUse = wrapClient
 		}
+		if len(header) > 0 {
```

Comment:
This `if` isn't necessary.

Reply:
Good catch! I'd had it written a different way earlier where I needed the guard, but forgot to remove it when I switched to iterating.